src/nvim/shada.c

```diff
@@ -1795,31 +1817,31 @@ static ShaDaWriteResult shada_pack_entry(msgpack_packer *const packer, ShadaEntr
   if (!max_kbyte || sbuf.size <= max_kbyte * 1024) {
     if (entry.type == kSDItemUnknown) {
       if (msgpack_pack_uint64(packer, entry.data.unknown_item.type) == -1) {
-        goto shada_pack_entry_error;
+        shada_pack_entry_error(spacker, sbuf);
+        return ret;
       }
     } else {
       if (msgpack_pack_uint64(packer, (uint64_t)entry.type) == -1) {
-        goto shada_pack_entry_error;
+        shada_pack_entry_error(spacker, sbuf);
+        return ret;
       }
     }
     if (msgpack_pack_uint64(packer, (uint64_t)entry.timestamp) == -1) {
-      goto shada_pack_entry_error;
+      shada_pack_entry_error(spacker, sbuf);
+      return ret;
     }
     if (sbuf.size > 0) {
       if ((msgpack_pack_uint64(packer, (uint64_t)sbuf.size) == -1)
           || (packer->callback(packer->data, sbuf.data,
                                (unsigned)sbuf.size) == -1)) {
-        goto shada_pack_entry_error;
+        shada_pack_entry_error(spacker, sbuf);
+        return ret;
       }
     }
   }
   msgpack_packer_free(spacker);
   msgpack_sbuffer_destroy(&sbuf);
   return kSDWriteSuccessfull;
-shada_pack_entry_error:
```

Comment:
This one is a conventional "goto error" style goto tho, which is commonly used in our codebase.

Reply:
ok